### PR TITLE
joplin-desktop: 1.8.5 -> 2.1.9

### PR DESCRIPTION
**Pull request by CheariX**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/joplin-desktop/default.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 let
   pname = "joplin-desktop";
-  version = "1.8.5";
+  version = "2.1.8";
```
Comment (mweinelt) [inline pkgs/applications/misc/joplin-desktop/default.nix:5]: https://github.com/laurent22/joplin/releases/tag/v2.1.9

**Comment by mweinelt** [comment]
Pinging #127408 for reviewers.

File: pkgs/applications/misc/joplin-desktop/default.nix (hunk 1 above)
Comment (CheariX) [reply pkgs/applications/misc/joplin-desktop/default.nix]: Sorry for the off-by-one :)

**Comment by emptyflask** [comment]
Tested successfully on NixOS and macOS.

**Pull request merged**